### PR TITLE
PP-7585 Connector client: switch more methods to use new baseclient

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Connector client: Updated more methods to use new base client

### CODE REVIEW

**Review by stephencdaly — COMMENTED** [review]
Looks good but left some comments

File: app/controllers/transactions/transaction-list.controller.js [hunk 1]
```diff
@@ -16,50 +16,61 @@ const client = new ConnectorClient(process.env.CONNECTOR_URL)
...
+    return error(req, res, 'Invalid search')
+  }
+
+  let transactions
+  let allCards
+
+  try {
+    transactions = await transactionService.search([accountId], filters.result, correlationId)
+  } catch (err) {
+    return error(req, res, 'Unable to retrieve list of transactions.')
```
[the remark this answers is not in the document]
Comment (kbottla) [reply app/controllers/transactions/transaction-list.controller.js]: 👍🏽 Updated to return next(err)

File: app/controllers/transactions/transaction-list.controller.js [hunk 2]
```diff
@@ -16,50 +16,61 @@ const client = new ConnectorClient(process.env.CONNECTOR_URL)
...
-              brand.selected = filters.result.brand.includes(brand.value)
-            })
-          }
-          model.clearRedirect = router.paths.transactions.index
-          model.isStripeAccount = req.account.payment_provider === 'stripe'
-          response(req, res, 'transactions/index', model)
-        })
-        .on('connectorError', () => error('Unable to retrieve card types.'))
+  try {
+    allCards = await client.getAllCardTypes(correlationId)
```
[the remark this answers is not in the document]
Comment (kbottla) [reply app/controllers/transactions/transaction-list.controller.js]: Updated!

File: app/services/clients/connector.client.js [hunk 3]
```diff
@@ -120,17 +118,17 @@ function _serviceNameUrlFor (gatewayAccountId, url) {
 }
 
 /** @private */
-var _getNotificationEmailUrlFor = function (accountID, url) {
+let _getNotificationEmailUrlFor = function (accountID, url) {
```
[the remark this answers is not in the document]
Comment (kbottla) [reply app/services/clients/connector.client.js]: 👍🏽 updated!

File: app/controllers/transactions/transaction-list.controller.js [hunk 4]
```diff
@@ -16,50 +16,56 @@ const client = new ConnectorClient(process.env.CONNECTOR_URL)
...
-                text: state.name,
-                selected: filters.result.selectedStates && filters.result.selectedStates.includes(state.name)
-              }
-            })
-          model.eventStates.unshift({ value: '', text: 'Any', selected: false })
+  if (!filters.valid) {
+    return error(req, res, 'Invalid search')
+  }
+
+  let result
```
Comment (stephencdaly) [inline app/controllers/transactions/transaction-list.controller.js:34]: I'd be tempted to put the try/catch around the whole of the rest of the function so we don't need to declare using `let`. But there isn't a problem with it this way.

**Review by stephencdaly — APPROVED** [review]
Looks good

**Pull request merged**